lib/connectors/memory.js

```diff
@@ -383,7 +399,9 @@ Memory.prototype._castPropertyValue = function(prop, val, props) {
       break;
     case 'ModelConstructor':
       for (var subProp in val) {
-        val[subProp] = this._castPropertyValue(subProp, val[subProp], propType.definition.properties);
```

Comment:
It's just a recursive check, since the property could itself be a model in need of having its own properties populated.